flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/batch/BatchExecPythonGroupWindowAggregate.java

```diff
@@ -0,0 +1,219 @@
+        this.grouping = grouping;
+        this.groupingSet = groupingSet;
+        this.aggCalls = aggCalls;
+        this.window = window;
+        this.inputTimeFieldIndex = inputTimeFieldIndex;
+        this.namedWindowProperties = namedWindowProperties;
+    }
+
+    @Override
+    protected Transformation<RowData> translateToPlanInternal(PlannerBase planner) {
+        final ExecNode<RowData> inputNode = (ExecNode<RowData>) getInputNodes().get(0);
+        final Transformation<RowData> inputTransform = inputNode.translateToPlan(planner);
+        final RowType inputRowType = (RowType) inputNode.getOutputType();
+        final RowType outputRowType = InternalTypeInfo.of(getOutputType()).toRowType();
+
+        final Tuple2<Long, Long> windowSizeAndSlideSize = WindowCodeGenerator.getWindowDef(window);
+        final TableConfig tableConfig = planner.getTableConfig();
+        final Configuration config = CommonPythonUtil.getConfig(planner.getExecEnv(), tableConfig);
+        int groupBufferLimitSize =
+                config.getInteger(ExecutionConfigOptions.TABLE_EXEC_WINDOW_AGG_BUFFER_SIZE_LIMIT);
+
+        OneInputTransformation<RowData, RowData> transform =
+                createPythonOneInputTransformation(
+                        inputTransform,
+                        inputRowType,
+                        outputRowType,
+                        groupBufferLimitSize,
+                        windowSizeAndSlideSize.f0,
+                        windowSizeAndSlideSize.f1,
+                        config);
+        if (CommonPythonUtil.isPythonWorkerUsingManagedMemory(config)) {
+            transform.declareManagedMemoryUseCaseAtSlotScope(ManagedMemoryUseCase.PYTHON);
+        }
+        return transform;
+    }
+
+    private OneInputTransformation<RowData, RowData> createPythonOneInputTransformation(
+            Transformation<RowData> inputTransform,
+            RowType inputRowType,
+            RowType outputRowType,
+            int maxLimitSize,
+            long windowSize,
+            long slideSize,
+            Configuration config) {
+        int[] namePropertyTypeArray =
+                Arrays.stream(namedWindowProperties)
+                        .mapToInt(
+                                p -> {
+                                    PlannerWindowProperty property = p.property();
+                                    if (property instanceof PlannerWindowStart) {
+                                        return 0;
+                                    }
+                                    if (property instanceof PlannerWindowEnd) {
+                                        return 1;
+                                    }
+                                    if (property instanceof PlannerRowtimeAttribute) {
+                                        return 2;
+                                    }
+                                    if (property instanceof PlannerProctimeAttribute) {
```

Comment:
There is the corresponding code for this in the original Scala class, could you double check that?

Reply:
Good Catch